src/azure-cli/azure/cli/command_modules/acr/custom.py

```diff
@@ -48,7 +49,21 @@ def acr_create(cmd,
     registry = Registry(location=location, sku=Sku(name=sku), admin_user_enabled=admin_enabled, tags=tags)
     if default_action:
         registry.network_rule_set = NetworkRuleSet(default_action=default_action)
-    return client.create(resource_group_name, registry_name, registry)
+    lro_poller = client.create(resource_group_name, registry_name, registry)
+    if workspace:
+        from azure.cli.core.commands import LongRunningOperation
+        from msrestazure.tools import is_valid_resource_id, resource_id
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        acr = LongRunningOperation(cmd.cli_ctx)(lro_poller)
+        if not is_valid_resource_id(workspace):
+            workspace = resource_id(subscription=get_subscription_id(cmd.cli_ctx),
+                                    resource_group=resource_group_name,
+                                    namespace='microsoft.OperationalInsights',
+                                    type='workspaces',
+                                    name=workspace)
```

Comment:
I wrote similar code recently, which transforms name to ID.  Maybe we can abstract a layer in core to handle this so that we don't need write same transformation code for every parameter everywhere.

Reply:
I gave it a thought, but felt it doesn't save lots of code except the subscription part. If that is what you meant, then yes I can add a core.util